proposals/memory-specializations/utilities.tex

```diff
@@ -0,0 +1,349 @@
+%!TEX root = std.tex
+\setcounter{chapter}{19}
+\rSec0[utilities]{General utilities library}
+
+\setcounter{section}{6}
+\rSec1[memory]{Memory}
+\synopsis{Header \tcode{<\added{experimental/ranges/}memory>} synopsis}
+
+\setcounter{subsection}{11}
+\rSec2[specialized.algorithms]{Specialized algorithms}
+
+\rSec3[uninitialized.construct.default]{uninitialized_default_construct}
+\begin{codeblock}
+namespace std { @\added{namespace experimental \{ namespace ranges \{ inline namespace v1 \{}@
+  template <@\changed{class ForwardIterator}{OutputIterator O, Sentinel<O> S}@>
+    @\added{requires DefaultConstructible<value_type_t<O>{>}() \&\&}@
+             @\added{Same<value_type_t<O>\&, reference_t<O>{>}}@
+    @\changed{void}{O}@ uninitialized_default_construct(@\changed{ForwardIterator}{O}@ first, @\changed{ForwardIterator}{S}@ last);
+\end{codeblock}
+
+\setcounter{Paras}{0}
+\pnum
+\effects Equivalent to:
+\begin{codeblock}
+        for (; first != last; ++first)
+          ::new(static_cast<void*>(addressof(first)))
+            @\changed{typename iterator_traits<ForwardIterator>::value_type}{value_type_t<O>}@;
+        @\added{return last;}@
+\end{codeblock}
+
+{\color{addclr}
+\begin{codeblock}
+  template <OutputRange Rng>
+    requires DefaultConstructible<value_type_t<iterator_t<Rng>>> &&
+             Same<value_type_t<iterator_t<O>>, reference_t<iterator_t<O>>
+  O uninitialized_default_construct(Rng&& rng);
+\end{codeblock}
+
+\pnum
+\effects Equivalent to:
+\begin{codeblock}
+        return last uninitialized_default_construct(begin(rng), end(rng));
+\end{codeblock}
+} %% \color{addclr}
+
+\begin{codeblock}
+  template <@\changed{class}{OutputIterator}@ @\changed{OutputIterator}{O}\removed{, class Size}@>
```

Comment:
Ah ok then `ForwardIterator` but not `OutputIterator` because that requires `Writable` which we don't need.